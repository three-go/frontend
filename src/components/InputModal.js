import React, { useState } from "react";

import { Modal, StyleSheet, Text, View, TextInput } from "react-native";

const InputModal = ({
  score,
  modalVisible,
  name,
  setName,
  onRegister,
  children,
}) => {
  return (
    <View style={styles.container}>
      <Modal transparent={true} visible={modalVisible}>
        <View style={styles.container}>
          <View style={styles.modalView}>
            <Text style={styles.modalText}>점수: {score}</Text>

            <Text style={styles.modalText}>
              스코어에 등록할 이름을 입력해 주세요 .
            </Text>

            <TextInput
              style={styles.inputStyle}
              placeholder="이름을 입력하세요."
              value={name}
              onChangeText={setName}
              onSubmitEditing={onRegister}
              returnKeyType="done"
            />

            {children}
          </View>
        </View>
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 20,
  },
  modalView: {
    alignItems: "center",
    margin: 20,
    padding: 35,
    borderRadius: 20,
    width: "80%",
    height: "30%",
    backgroundColor: "white",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  inputStyle: {
    marginTop: 10,
    marginBottom: 20,
    borderBottomWidth: 1,
    borderColor: "#2196F3",
    width: "80%",
    textAlign: "center",
  },
  button: {
    padding: 10,
    borderRadius: 20,
    elevation: 2,
    backgroundColor: "#2196F3",
  },
  textStyle: {
    color: "white",
    textAlign: "center",
    fontWeight: "bold",
  },
  modalText: {
    marginTop: 15,
    marginBottom: 15,
    textAlign: "center",
  },
});

export default InputModal;
