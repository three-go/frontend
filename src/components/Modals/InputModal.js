import React from "react";

import {
  Modal,
  StyleSheet,
  Text,
  View,
  TextInput,
  KeyboardAvoidingView,
  Platform,
} from "react-native";
import PropTypes from "prop-types";

import { colors } from "../../common";

const InputModal = ({
  score,
  modalVisible,
  name,
  setName,
  onRegister,
  children,
}) => {
  return (
    <Modal transparent={true} visible={modalVisible}>
      <KeyboardAvoidingView
        behavior={Platform.OS === "ios" ? "padding" : undefined}
        style={styles.container}
      >
        <View style={styles.modalView}>
          <Text style={styles.modalTitle}>축하합니다 !</Text>
          <Text style={styles.modalScore}>점수: {score}</Text>

          <Text style={styles.modalText}>
            점수 기록에 사용할 이름을
            {"\n"}
            입력해 주세요.
          </Text>

          <TextInput
            style={styles.inputStyle}
            placeholder="이름을 입력하세요."
            value={name}
            onChangeText={setName}
            onSubmitEditing={onRegister}
            returnKeyType="done"
          />

          {children}
        </View>
      </KeyboardAvoidingView>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 20,
  },
  modalView: {
    justifyContent: "space-between",
    alignItems: "center",
    width: 350,
    height: 350,
    margin: 20,
    padding: 15,
    borderWidth: 5,
    borderRadius: 20,
    borderColor: colors.ivory,
    backgroundColor: colors.gray,
    elevation: 5,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
  },
  modalTitle: {
    color: colors.ivory,
    fontSize: 22,
  },
  modalScore: {
    color: colors.ivory,
    fontSize: 30,
    fontWeight: "bold",
  },
  modalText: {
    color: colors.ivory,
    textAlign: "center",
    fontSize: 18,
  },
  inputStyle: {
    borderBottomWidth: 1,
    borderColor: colors.ivory,
    color: colors.ivory,
    fontSize: 18,
    textAlign: "center",
  },
  button: {
    padding: 10,
    borderRadius: 20,
    backgroundColor: colors.tealGreen,
    elevation: 2,
  },
  textStyle: {
    color: colors.ivory,
    fontWeight: "bold",
    textAlign: "center",
  },
});

export default InputModal;

InputModal.propTypes = {
  score: PropTypes.number.isRequired,
  modalVisible: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  setName: PropTypes.func.isRequired,
  onRegister: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired,
};
