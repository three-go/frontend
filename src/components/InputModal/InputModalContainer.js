import React, { useState } from "react";

import { Keyboard } from "react-native";

import { SmallButton, InputModal } from "../../components";

const InputModalContainer = () => {
  const [modalVisible, setModalVisible] = useState(true);
  const [name, setName] = useState("");

  // 유저 스코어 로직 가져오는 로직 필요

  const onRegister = (name) => {
    // 유저 스코어 등록 로직 필요
    setModalVisible(false);
    setName("");
    Keyboard.dismiss();
  };

  return (
    <InputModal
      score="200"
      modalVisible={modalVisible}
      name={name}
      setName={setName}
      onRegister={onRegister}
    >
      <SmallButton content="등록" color="#2196F3" onPress={onRegister} />
    </InputModal>
  );
};

export default InputModalContainer;
