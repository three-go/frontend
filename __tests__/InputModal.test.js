import React from "react";

import { render, fireEvent } from "@testing-library/react-native";
import { Text } from "react-native";

import InputModal from "../src/components/Modals/InputModal";

describe("InputModal", () => {
  it("InputModal render check", () => {
    const score = 100;
    const modalVisible = true;
    const name = "test";
    const mockSetName = jest.fn();
    const mockOnRegister = jest.fn();

    const component = render(
      <InputModal
        score={score}
        modalVisible={modalVisible}
        name={name}
        setName={mockSetName}
        onRegister={mockOnRegister}
      >
        <Text>Testing</Text>
      </InputModal>
    );

    expect(component.getByText(new RegExp(score, "i"))).toBeTruthy();
    expect(component.getByText(/점수 기록에 사용할 이름을/i)).toBeTruthy();
    expect(component.getByPlaceholderText("이름을 입력하세요.")).toBeTruthy();
    expect(component.getByText("Testing")).toBeTruthy();
    expect(component.toJSON).toMatchSnapshot();
  });
  it("InputModal submit textInput", () => {
    const score = 100;
    const modalVisible = true;
    const name = "test";
    const mockSetName = jest.fn();
    const mockOnRegister = jest.fn();

    const component = render(
      <InputModal
        score={score}
        modalVisible={modalVisible}
        name={name}
        setName={mockSetName}
        onRegister={mockOnRegister}
      >
        <Text>Testing</Text>
      </InputModal>
    );

    const textInput = component.getByPlaceholderText("이름을 입력하세요.");
    fireEvent(textInput, "submitEditing");

    expect(mockOnRegister.mock.calls.length).toBe(1);
  });
});
