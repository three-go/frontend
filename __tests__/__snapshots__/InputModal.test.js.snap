// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputModal InputModal render check 1`] = `[Function]`;
